tests/io/test_kedro_data_catalog.py

```diff
@@ -0,0 +1,658 @@
+            with pytest.raises(DatasetError, match=re.escape(pattern)):
+                data_catalog.confirm(dataset_name)
+
+    class TestDataCatalogVersioned:
+        def test_from_correct_config_versioned(self, correct_config, dummy_dataframe):
+            """Test load and save of versioned data sets from config"""
+            correct_config["catalog"]["boats"]["versioned"] = True
+
+            # Decompose `generate_timestamp` to keep `current_ts` reference.
+            current_ts = datetime.now(tz=timezone.utc)
+            fmt = (
+                "{d.year:04d}-{d.month:02d}-{d.day:02d}T{d.hour:02d}"
+                ".{d.minute:02d}.{d.second:02d}.{ms:03d}Z"
+            )
+            version = fmt.format(d=current_ts, ms=current_ts.microsecond // 1000)
+
+            catalog = KedroDataCatalog.from_config(
+                **correct_config,
+                load_versions={"boats": version},
+                save_version=version,
+            )
+
+            catalog.save("boats", dummy_dataframe)
+            path = Path(correct_config["catalog"]["boats"]["filepath"])
+            path = path / version / path.name
+            assert path.is_file()
+
+            reloaded_df = catalog.load("boats")
+            assert_frame_equal(reloaded_df, dummy_dataframe)
+
+            reloaded_df_version = catalog.load("boats", version=version)
+            assert_frame_equal(reloaded_df_version, dummy_dataframe)
+
+            # Verify that `VERSION_FORMAT` can help regenerate `current_ts`.
+            actual_timestamp = datetime.strptime(
+                catalog.datasets["boats"].resolve_load_version(),
+                VERSION_FORMAT,
+            )
+            expected_timestamp = current_ts.replace(
+                microsecond=current_ts.microsecond // 1000 * 1000, tzinfo=None
+            )
+            assert actual_timestamp == expected_timestamp
+
+        @pytest.mark.parametrize("versioned", [True, False])
+        def test_from_correct_config_versioned_warn(
+            self, caplog, correct_config, versioned
+        ):
+            """Check the warning if `version` attribute was added
+            to the data set config"""
+            correct_config["catalog"]["boats"]["versioned"] = versioned
+            correct_config["catalog"]["boats"]["version"] = True
+            KedroDataCatalog.from_config(**correct_config)
+            log_record = caplog.records[0]
+            expected_log_message = (
+                "'version' attribute removed from data set configuration since it "
+                "is a reserved word and cannot be directly specified"
+            )
+            assert log_record.levelname == "WARNING"
+            assert expected_log_message in log_record.message
+
+        def test_from_correct_config_load_versions_warn(self, correct_config):
+            correct_config["catalog"]["boats"]["versioned"] = True
+            version = generate_timestamp()
+            load_version = {"non-boart": version}
```

Comment:
Nit: but this looks like it's always been a typo ⛵ 
```suggestion
            load_version = {"non-boat": version}
```